process/smartContract/hooks/blockChainHook_test.go

```diff
@@ -54,98 +61,185 @@ func createMockVMAccountsArguments() hooks.ArgBlockChainHook {
 	return arguments
 }
 
-func TestNewBlockChainHookImpl_NilAccountsAdapterShouldErr(t *testing.T) {
-	t.Parallel()
-
-	args := createMockVMAccountsArguments()
-	args.Accounts = nil
-	bh, err := hooks.NewBlockChainHookImpl(args)
-
-	assert.Nil(t, bh)
-	assert.Equal(t, process.ErrNilAccountsAdapter, err)
-}
-
-func TestNewBlockChainHookImpl_NilPubkeyConverterShouldErr(t *testing.T) {
-	t.Parallel()
-
-	args := createMockVMAccountsArguments()
-	args.PubkeyConv = nil
-	bh, err := hooks.NewBlockChainHookImpl(args)
-
-	assert.Nil(t, bh)
-	assert.Equal(t, process.ErrNilPubkeyConverter, err)
-}
-
-func TestNewBlockChainHookImpl_NilStorageServiceShouldErr(t *testing.T) {
-	t.Parallel()
-
-	args := createMockVMAccountsArguments()
-	args.StorageService = nil
-	bh, err := hooks.NewBlockChainHookImpl(args)
-
-	assert.Nil(t, bh)
-	assert.Equal(t, process.ErrNilStorage, err)
+func createContractCallInput(function string, sender, receiver []byte) *vmcommon.ContractCallInput {
+	return &vmcommon.ContractCallInput{
+		Function:      function,
+		RecipientAddr: receiver,
+		VMInput: vmcommon.VMInput{
+			CallerAddr: sender,
+		},
+	}
 }
 
-func TestNewBlockChainHookImpl_NilBlockChainShouldErr(t *testing.T) {
+func TestNewBlockChainHookImpl_NilAccountsAdapterShouldErr(t *testing.T) {
```

Comment:
might rename the test just to `TestNewBlockChainHookImpl`

Reply:
Right, missed it, thanks !